project-codeflare/codeflare-sdk
# Updating training script in basic_interactive notebook

# Issue link
[RHOAIENG-6475](https://issues.redhat.com/browse/RHOAIENG-6475)

# What changes have been made
When running GPU utilising workloads using the basic_interactive demo notebook I encountered a lot of errors while running the training script provided in the notebook. As it was added more than a year ago and may be out of date I have updated the notebook to use ray train using these examples: [Text classification](https://huggingface.co/docs/transformers/tasks/sequence_classification ), [Hugging Face Transformers + Ray Train ](https://docs.ray.io/en/latest/train/getting-started-transformers.html)

I have also updated the hf_interactive notebook which was using the same training script.

# Verification steps
 - Install RHOAI with the CodeFlare components set to `Managed`
 - Ensure that your cluster has GPU nodes as required.
 - Within a data science project workbench clone this branch and install codeflare sdk
```
git clone https://github.com/Fiona-Waters/codeflare-sdk.git -b udpate-notebook-script
cd codeflare-sdk
pip install poetry
poetry install
```
 - Run the basic_interactive notebook and ensure that it works as expected.
 - Note that aws credentials are required and you will need to create and use an s3 bucket while running the training script. See this [PR](https://github.com/project-codeflare/codeflare-sdk/pull/563/files) for instructions.

## Checks
- Testing Strategy
